### PR TITLE
[ISSUE-2903] - Deprecating chat domain leave channel

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
Solves issue #2903 

### 🎯 Goal

Deprecate ChatDomain.leaveChannel

### 🧪 Testing

Remove some channels

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF
giphy
_Sorry for the old method D=_

### CODE REVIEW

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt [hunk 1]
```diff
@@ -92,7 +93,9 @@ public class ChannelsViewModel(
     }
 
     public fun leaveChannel(channel: Channel) {
```
Comment (MarinTolic) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:95]: Same here, if we could touch upon kdocs whenever we make changes that would make it easier on us to have consistent code documentation in the whole project.

**Review by MarinTolic — APPROVED** [review]
There were some problems with the consistency of leaving channels. However @leandroBorgesFerreira  has already created (this PR)[https://github.com/GetStream/stream-chat-android/pull/2933] to tackle this issue.

**Note:** Compose Sample app currently does not offer the option to leave channels so that could not be tested in it.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [  ] ~~UI Changes correct (before & after images)~~
- [ ] ~~Bugs validated (bugfixes)~~
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] ~~All code we touched has new or updated KDocs~~

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt [hunk 2]
```diff
@@ -1304,6 +1304,13 @@ public class ChatClient internal constructor(
         )
     }
 
+    /**
+     * Method to remove members of a given channel.
+     *
+     * @param channelType The channel type. ie messaging.
+     * @param channelId The channel id. ie 123.
+     * @param members The list of the members to be removed
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt:1312]:
```suggestion
     * @param members The list of the members to be removed.
```

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt [hunk 3]
```diff
@@ -164,13 +164,20 @@ public class ChannelListViewModel(
         }
     }
 
+    /**
+     * This method makes the current user to leave the channel of the param
```
Comment (adasiewiczr) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt:168]:
```suggestion
     * Removes the current user from the channel.
```

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt [hunk 4]
```diff
@@ -164,13 +164,20 @@ public class ChannelListViewModel(
...
+     *
+     * @param channel - The channel that the current user will leave
```
Comment (adasiewiczr) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/viewmodel/ChannelListViewModel.kt:170]:
```suggestion
     * @param channel - The channel that the current user will leave.
```

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt [hunk 5]
```diff
@@ -91,8 +91,15 @@ public class ChannelsViewModel(
         }
     }
 
+    /**
+     * This method makes the current user to leave the channel of the param
```
Comment (adasiewiczr) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:95]:
```suggestion
     * Removes the current user from the channel.
```

File: stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt [hunk 6]
```diff
@@ -91,8 +91,15 @@ public class ChannelsViewModel(
...
+     *
+     * @param channel - The channel that the current user will leave
```
Comment (adasiewiczr) [inline stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt:97]:
```suggestion
     * @param channel - The channel that the current user will leave.
```

**Review by adasiewiczr — APPROVED** [review]
Added small KDocs suggestions

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**